File: 01-week-1--starter-algorithms/02-day-3--fibonacci-series/javascript/fibonacci.js
function fibonacci(num) {
  let output = [0]
  if (num>=1) {
    output.push(1)
    for (i=1;i<num;i++) {
      output.push(parseInt(output[i-1]) + parseInt(output[i]))
    }
  }
  return output[output.length - 1]
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("");

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));
}

module.exports = fibonacci;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: 01-week-1--starter-algorithms/02-day-3--fibonacci-series/ruby/fibonacci.rb
# Given a number (n), we're gonna calculate the nth value from the beginning of the Fibonacci Sequence, a logical order that adds that last two values to each other to produce the next value

def fibonacci(num)
  output = [0]
  if num >= 1
    output << 1
    i = 1
    while i < num
      output << (output[i - 1] + output[i])
      i += 1
    end
  end
  output.last
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts "=>", fibonacci(0)

  puts

  puts "Expecting: 1"
  puts "=>", fibonacci(2)

  puts

  puts "Expecting: 55"
  puts "=>", fibonacci(10)

  # Don't forget to add your own!
end

# def fibonacci(num)
#   output = [0]
#   if num >= 1
#     output << 1
#
#
